contract/src/main/kotlin/com/ritense/valtimo/contract/client/HostDockerInternalRestClientCustomizer.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.contract.client
+
+import com.ritense.valtimo.contract.annotation.SkipComponentScan
+import com.ritense.valtimo.contract.io.FindReplaceInputStream
+import org.springframework.boot.web.client.RestClientCustomizer
+import org.springframework.core.env.Environment
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpRequest
+import org.springframework.http.MediaType.APPLICATION_JSON
+import org.springframework.http.client.ClientHttpRequestExecution
+import org.springframework.http.client.ClientHttpRequestInterceptor
+import org.springframework.http.client.ClientHttpResponse
+import org.springframework.http.client.support.HttpRequestWrapper
+import org.springframework.stereotype.Component
+import org.springframework.web.client.RestClient
+import java.net.URI
+
+@Component
+@SkipComponentScan
+class HostDockerInternalRestClientCustomizer(
+    environment: Environment,
+) : RestClientCustomizer, ClientHttpRequestInterceptor {
+
+    private val isDevelopment = environment.activeProfiles.contains("dev")
```

Comment:
Please don't check the profiles within a bean instance. That should be part of the app Configuration where the bean is enabled on whatever conditions needed.

Reply:
Changed